### PR TITLE
[Work in Progress] docs: API > `apollo-server`.

### PR DESCRIPTION
**Pull request by abernix**
Beginnings of `apollo-server` API docs, but not the end.

### CODE REVIEW

**Comment by abernix** [comment]
Mainly soliciting feedback about the layout of the API page, seen in this PR's Netlify preview [here](https://deploy-preview-997--apollo-server-docs.netlify.com/docs/apollo-server/api/apollo-server.html).

I think if we resolve https://github.com/meteor/meteor-theme-hexo/issues/58, it will be much better, since the `<h4>` and `<h5>` are not properly proportioned against `<h1>`, `<h2>` and `<h3>`.

File: docs/source/api/apollo-server.md [hunk 1]
```diff
@@ -2,12 +2,149 @@
...
+This API reference documents the exports from the `apollo-server`, and `apollo-server/{variant}` packages (e.g. `apollo-server/express`).  Both the root module and the module sub-paths export the same functionality.
 
-## `ApolloServer` constructor
+## `ApolloServer`
+
+The `ApolloServer` class is used to create an instance of Apollo Server.  An instance of Apollo Server will only be available when `listen` is called.
+
+If using Apollo Server as middleware in an existing project, it is also necessary to call `applyMiddleware`.
+
+### `new ApolloServer(options)`: <`ApolloServer`>
```
[the remark this answers is not in the document]
Comment (abernix) [reply docs/source/api/apollo-server.md]: 👍 I clearly hadn't considered the way this would appear in the sidebar menu (nor did I look at the Netlify preview).

I'm going to try going with `ApolloServer` and dropping this into a `constructor(options)` within it, then put "Usage" under that.

File: docs/source/api/apollo-server.md [hunk 2]
```diff
@@ -2,12 +2,149 @@
...
+
+    Type: `Object`
+
+    This should be a map of resolvers for the types defined in `typeDefs`.  The key should be the type name and the value should be a `Function` to be executed for that type.
+
+#### Returns
+
+`ApolloServer`
+
+### `ApolloServer.prototype.listen(options)`: `Promise`
```
[the remark this answers is not in the document]
Comment (abernix) [reply docs/source/api/apollo-server.md]: Hmm, `ApolloServer.listen` doesn't portray that this is an instance method, not a class method; I think it's semantically incorrect to represent it as `ApolloServer.listen`.

I'm not a huge an of `#` being used as shorthand for `prototype`, especially given the stage 3 https://github.com/tc39/proposal-private-methods proposal but what about `ApolloServer#listen`?

File: docs/source/api/apollo-server.md [hunk 3]
```diff
@@ -2,12 +2,149 @@
 title: apollo-server
```
[the remark this answers is not in the document]
Comment (abernix) [reply docs/source/api/apollo-server.md]: As these are API references, I was thinking "Apollo Engine" might actually be the outlier here, and should be changed to `apollo-engine` (I hadn't touched that page yet, so it hadn't been set)?  Furthermore, I think we may be introducing `graphql-tools` here as well.  In other words, the API references are intending to say "These are how we use these specific modules" not referencing them by their product name.

Basically, I'm afraid that using "Apollo Server" here would be like saying "Lodash Sorted Last Index By" when the API I'm looking for is `lodash/sortedLastIndexBy`.

File: docs/source/api/apollo-server.md [hunk 4]
```diff
@@ -2,12 +2,149 @@
...
 ---
 
-> WIP / TODO.
+This API reference documents the exports from the `apollo-server`, and `apollo-server/{variant}` packages (e.g. `apollo-server/express`).  Both the root module and the module sub-paths export the same functionality.
 
-## `ApolloServer` constructor
+## `ApolloServer`
+
+The `ApolloServer` class is used to create an instance of Apollo Server.  An instance of Apollo Server will only be available when `listen` is called.
```
[the remark this answers is not in the document]
Comment (abernix) [reply docs/source/api/apollo-server.md]: Yeah, this is all sorts of wrong.

File: docs/source/api/apollo-server.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (abernix) [reply docs/source/api/apollo-server.md]: Maybe the correct answer is just to just say `listen` and to prefix any class methods with `static ` (`static`-space).

**Comment by abernix** [comment]
I'm beginning to think that, instead of writing this out in this way, we should just get the JSDoc declarations under way within Apollo Server.
